Домашнее задание спринт 15

NekhoroshkinaElena opened the pull request.

NekhoroshkinaElena committed:
fix: исправить баг (при клике одновременно на два элемента списка открывается сразу 2 activity)
fix: исправить баг (при быстром стирании текста с помощью клавиатуры, не исчезают сообщения о неправильном запросе или отсутствии подключения к интернету)

NekhoroshkinaElena committed:
init commit

taikonavt commented on app/src/main/java/com/example/playlistmaker/data/dto/TrackDto.kt:
```diff
@@ -0,0 +1,14 @@
+    val trackTimeMillis: String,
+    val artworkUrl100: String,
+    val releaseDate: String,
+    val primaryGenreName: String,
```
при запросе nirvana происходит падение с ошибкой 
java.lang.NullPointerException: Parameter specified as non-null is null: method com.example.playlistmaker.domain.models.Track.<init>, parameter primaryGenreName

taikonavt commented on app/src/main/java/com/example/playlistmaker/presentation/search/SearchActivity.kt:
```diff
@@ -1,5 +1,6 @@
-package com.example.playlistmaker
+package com.example.playlistmaker.presentation.search
 
+import Creator
```
обычно файлы не хранят в этой папке. Вроде все работает, но не понятно к чему относится класс Creator (к какой то библиотеке или к твоему приложению). Поэтому лучше не клади файлы выше com.example.playlistmaker
я бы положил этот класс в util или в com.example.playlistmaker

taikonavt commented on app/src/main/java/com/example/playlistmaker/domain/models/TracksListResponse.kt:
```diff
@@ -0,0 +1,3 @@
+package com.example.playlistmaker.domain.models
+
+data class TracksListResponse(val tracks: List<Track>, val codeResponse: Int)
```
когда мы разбиваем классы на слои, мы хотим разделить получение/хранение данных (repository) и бизнес логику (interactor). Это позволяет уменьшить связность приложения и его будет проще изменять. Когда интерактор получает данные из репо, то ему должно быть все равно откуда приходят данные: из базы данных, из сети, из sharedPref'ов - важна только бизнес логика. Поэтому в интеракторе мы максимально убираем слова и действия типа: saveToSharedPrefs, responseFromNetwork, requestToServer и пр., которые указывают на способ хранения и получения данных. 
Лучше переименовать этот класс во что то типа TracksListResult например

NekhoroshkinaElena committed:
fix: исправить баг (приложение падает при запросе nirvana )
fix: переложить Creator.kt в другой пакет
fix: переименовать класс TracksListResponse.kt
